.infrastructure/deployment.yml

```diff
@@ -0,0 +1,52 @@
+apiVersion: apps/v1
+kind: Deployment
+
+metadata:
+  name: todoapp
+  namespace: mateapp
+
+spec:
+  selector:
+   matchLabels:
+     app: todoapp
+  
+  replicas: 2
+  
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 1
+  
+  template:
+    name: todoapp
```

Comment:
The `name` field is not valid within the `template` section of a Deployment. You should remove this line as the `name` is not a valid field for Pod templates.